lib/config-generator.js

```diff
@@ -324,20 +324,25 @@ class ConfigGenerator {
          *  * https://github.com/webpack/webpack.js.org/issues/652#issuecomment-273324529
          *  * https://webpack.js.org/guides/caching/#deterministic-hashes
          */
-        // when not using versioning, none of this is important
+        let moduleNamePlugin;
+        if (this.webpackConfig.isProduction()) {
+            // shorter, and obfuscated module ids (versus NamedModulesPlugin)
+            // makes the final assets *slightly* larger, but prevents contents
+            // from sometimes changing when nothing really changed
+            moduleNamePlugin = new webpack.HashedModuleIdsPlugin();
+        } else {
+            // human-readable module names, helps debug in HMR
+            // enable always when ! production for consistency
+            moduleNamePlugin = new webpack.NamedModulesPlugin();
+        }
+
+        if (moduleNamePlugin) {
+            plugins.push(moduleNamePlugin);
```

Comment:
you could put this directly above instead of using a `moduleNamePlugin` variable.